src/families/hedera/js-scanAccounts.ts

```diff
@@ -0,0 +1,126 @@
+import { CurrencyNotSupported } from "@ledgerhq/errors";
+import { Observable } from "rxjs";
+import {
+  Account,
+  CryptoCurrency,
+  DerivationMode,
+  DeviceId,
+  ScanAccountEvent,
+  SyncConfig,
+} from "../../types";
+import { open, close } from "../../hw";
+import {
+  getDerivationModesForCurrency,
+  getDerivationScheme,
+  getSeedIdentifierDerivation,
+  runDerivationScheme,
+} from "../../derivation";
+import getAddress from "../../hw/getAddress";
+import BigNumber from "bignumber.js";
+import {
+  getAccountPlaceholderName,
+  emptyHistoryCache,
+} from "../../account";
+import { getAccountsForPublicKey } from "./api/mirror";
+import { log } from "@ledgerhq/logs";
+
+export default function scanAccounts({
+  currency,
+  deviceId,
+  scheme,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  syncConfig,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  preferredNewAccountScheme,
+}: {
+  currency: CryptoCurrency;
+  deviceId: DeviceId;
+  scheme?: DerivationMode | null | undefined;
+  syncConfig: SyncConfig;
+  preferredNewAccountScheme?: DerivationMode;
+}): Observable<ScanAccountEvent> {
+  var derivationMode = scheme ?? getDerivationModesForCurrency(currency)[0];
+
+  return new Observable((o) => {
```

Comment:
In future, it will be preferrable to use jsHelpers which should help you generalizing the 'scanAccounts' and 'sync'. Is there any reason it was not used yet?